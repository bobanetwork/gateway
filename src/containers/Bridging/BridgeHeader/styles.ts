import { SettingsOutlined } from '@mui/icons-material'
import styled from 'styled-components'

export const BridgeHeaderWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export const IconWrapper = styled.div<{ inline?: boolean }>`
  padding: 8px;
  height: 32px;
  width: 32px;
  border-radius: 50%;
  cursor: pointer;
  display: ${({ inline }) => (inline ? 'inline-flex' : 'flex')};
  justify-content: center;
  align-items: center;
  &:hover {
    background: ${({ theme: { name, colors } }) =>
      name === 'light' ? colors.gray[400] : colors.gray[300]};
  }
`

export const GearIcon = styled(SettingsOutlined)`
  cursor: pointer;
`

export const LabelStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 138.3%;
`

export const ValueStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 138.3%;
`
