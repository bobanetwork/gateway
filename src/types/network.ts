export type NetworkSwitchState =
  | 'idle'
  | 'confirming'
  | 'switching'
  | 'success'
  | 'failed'
  | 'chain-not-found'

export interface NetworkSwitchError {
  code?: number;
  message: string;
}
