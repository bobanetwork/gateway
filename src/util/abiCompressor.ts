/**
 * Script to prepare the human readable abi from the json.
 * based on the functions used out of the abi in network service.
 *
 * to prepare abi execute like
 * ```sh
 *  node ./util/abiCompressor.ts
 * ```
 *
 *
 */

import { ethers } from 'ethers'
import * as fs from 'fs'

const smart_contract_abi_paths = [
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/messaging/L2StandardBridge.sol/L2StandardBridge.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/standards/L2StandardERC20.sol/L2StandardERC20.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/predeploys/OVM_GasPriceOracle.sol/OVM_GasPriceOracle.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DiscretionaryExitFee.sol/DiscretionaryExitFee.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L1LiquidityPool.sol/L1LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/Teleportation.sol/Teleportation.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L2LiquidityPool.sol/L2LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/BobaFixedSavings.sol/BobaFixedSavings.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance-token/BOBA.sol/BOBA.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegate.sol/GovernorBravoDelegate.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegator.sol/GovernorBravoDelegator.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/L2BillingContract.sol/L2BillingContract.json',
  './src/deployment/contracts/L1ERC20.json',
  './src/deployment/contracts/NFTMonsterV2.json',
  './src/deployment/contracts/AuthenticatedFaucet.json',
  './src/deployment/contracts/WAGMIv0.json',
  './src/deployment/contracts/WAGMIv1.json',
  './src/deployment/contracts/crosschain/EthBridge.json',
  './src/deployment/contracts/crosschain/L2StandardERC20.json',
  './src/deployment/contracts/crosschain/LZEndpointMock.json',
]

const networkServicesStr = fs.readFileSync(
  './src/services/networkService.ts',
  'utf8'
)

const compress = () => {
  for (const path of smart_contract_abi_paths) {
    const rawData = fs.readFileSync(path, 'utf8')
    const abiObject = JSON.parse(rawData)
    const name = abiObject.contractName
    let abi = abiObject.abi
    abi = abi.filter((component) => {
      if (component['type'] === 'function' && component['name']) {
        if (networkServicesStr.includes(component['name'])) {
          return true
        }
        return false
      }
    })
    const abiInterface = new ethers.utils.Interface(abi)
    const humanReadableABI = abiInterface.format(ethers.utils.FormatTypes.full)
    const stringifiedHumanReadableABI = JSON.stringify(humanReadableABI)
    const result = `const ${name}ABI = ${stringifiedHumanReadableABI}\n export default ${name}ABI`
    fs.writeFile(`./src/services/abi/${name}.abi.ts`, result, () => {
      console.log(`wrote ${name}`)
    })
  }
}

;(() => {
  compress()
})()
