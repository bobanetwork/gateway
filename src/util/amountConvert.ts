/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js'
import { BigNumberish } from 'ethers'

export const logAmount = (
  amount: BigNumberish,
  power: number,
  truncate = 0
) => {
  // toString as some methods seem to supply BigNumber
  const x = new BigNumber(amount?.toString())
  const exp = new BigNumber(10).pow(power)

  const calculated = x.div(exp)

  if (truncate > 0) {
    return calculated.toFixed(truncate)
  } else {
    return calculated.toFixed()
  }
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export const powAmount = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export const toWei_String = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

const lookupMap = {
  ETH: { provider: 'ethereum' },
  BOBA: { provider: 'boba-network' },
  OLO: { provider: 'oolongswap' },
  OMG: { provider: 'omisego' },
  USDC: { provider: 'usd-coin' },
  BNB: { provider: 'binancecoin' },
  tBNB: { provider: 'binancecoin' },
}

export const amountToUsd = (amount: number, lookupPrice, token) => {
  const { symbol } = token
  if (!symbol) {
    return 0
  }
  const currencyInfo = lookupMap[symbol]
  if (!!currencyInfo && !!lookupPrice[currencyInfo.provider]) {
    return amount * lookupPrice[currencyInfo.provider].usd
  } else if (!!lookupPrice[symbol.toLowerCase()]) {
    return amount * lookupPrice[symbol.toLowerCase()].usd
  } else {
    return 0
  }
}

export const formatLargeNumber = (num: number) => {
  const exp = Math.floor(Math.log10(Math.abs(num)) / 3)
  const scaledNum = Math.abs(num) / Math.pow(10, exp * 3)
  const formattedNumber = scaledNum.toFixed(2)
  const exponent = ['', 'k', 'M', 'B', 'T'][exp] || ''
  return isNaN(Number(formattedNumber)) ? '0' : formattedNumber + exponent
}
