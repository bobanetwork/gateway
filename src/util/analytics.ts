import ReactGA from 'react-ga4'
import { GA4_MEASUREMENT_ID } from './constant'

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID?.toString())
  }
}

export const sendPageView = (path: string) => {
  ReactGA.send({
    hitType: 'pageview',
    page: path,
    title: path,
  })
}

export const trackClick = (category: string, action: string, label: string) => {
  ReactGA.event({
    category,
    action,
    label,
  })
}

export default {
  init,
  sendPageView,
  trackClick,
}
