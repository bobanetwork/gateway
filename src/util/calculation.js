/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair. 
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

var accAdd = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = (accMul(num1, times) + accMul(num2, times)) / times;
    return getCorrectResult("add", num1, num2, result);
};

var accSub = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = Number((accMul(num1, times) - accMul(num2, times)) / times);
    return getCorrectResult("sub", num1, num2, result);
};

var accDiv = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var t1 = 0, t2 = 0, dec1, dec2;
    try { t1 = countDecimals(num1); } catch (e) { }
    try { t2 = countDecimals(num2); } catch (e) { }
    dec1 = convertToInt(num1);
    dec2 = convertToInt(num2);
    var result = accMul((dec1 / dec2), Math.pow(10, t2 - t1));
    return getCorrectResult("div", num1, num2, result);
};

var accMul = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var times = 0, s1 = num1.toString(), s2 = num2.toString();
    try { times += countDecimals(s1); } catch (e) { }
    try { times += countDecimals(s2); } catch (e) { }
    var result = convertToInt(s1) * convertToInt(s2) / Math.pow(10, times);
    return getCorrectResult("mul", num1, num2, result);
};

var countDecimals = function(num) {
    var len = 0;
    try {
        num = Number(num);
        var str = num.toString().toUpperCase();
        if (str.split('E').length === 2) { // scientific notation
            var isDecimal = false;
            if (str.split('.').length === 2) {
                str = str.split('.')[1];
                if (parseInt(str.split('E')[0]) !== 0) {
                    isDecimal = true;
                }
            }
            let x = str.split('E');
            if (isDecimal) {
                len = x[0].length;
            }
            len -= parseInt(x[1]);
        } else if (str.split('.').length === 2) { // decimal
            if (parseInt(str.split('.')[1]) !== 0) {
                len = str.split('.')[1].length;
            }
        }
    } catch(e) {
        throw e;
    } finally {
        if (isNaN(len) || len < 0) {
            len = 0;
        }
        return len;
    }
};

var convertToInt = function(num) {
    num = Number(num);
    var newNum = num;
    var times = countDecimals(num);
    var temp_num = num.toString().toUpperCase();
    if (temp_num.split('E').length === 2) {
        newNum = Math.round(num * Math.pow(10, times));
    } else {
        newNum = Number(temp_num.replace(".", ""));
    }
    return newNum;
};

var getCorrectResult = function(type, num1, num2, result) {
    var temp_result = 0;
    switch (type) {
        case "add":
            temp_result = num1 + num2;
            break;
        case "sub":
            temp_result = num1 - num2;
            break;
        case "div":
            temp_result = num1 / num2;
            break;
        case "mul":
            temp_result = num1 * num2;
            break;
        default:
            break;
    }
    if (Math.abs(result - temp_result) > 1) {
        return temp_result;
    }
    return result;
};


export {
  accAdd,
  accSub,
  accDiv,
  accMul,
  countDecimals,
  convertToInt,
  getCorrectResult,
}