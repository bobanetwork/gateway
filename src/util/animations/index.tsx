import React, { useState, useEffect } from 'react'
import { TConductorInstance } from 'react-canvas-confetti/dist/types'
import Realistic from 'react-canvas-confetti/dist/presets/realistic'
import { TCanvasConfettiAnimationOptions } from 'react-canvas-confetti/dist/types/normalization'
import styled from 'styled-components'
import dayjs from 'dayjs'

const CanvaContainer = styled.div`
  display: block;
  z-index: 9999;
  position: fixed;
`
const count = 200
const defaults = {
  origin: { y: 0.7 },
}

export const Confetti = ({ size, particles, spreads }: any) => {
  const [conductor, setConductor] = useState<TConductorInstance>()

  const onInit = ({ conductor }: { conductor: TConductorInstance }) => {
    setConductor(conductor)
  }

  const onOnce = () => {
    conductor?.shoot()
  }

  const PARTICLES = {
    heart: [
      {
        type: 'path',
        path: 'M167 72c19,-38 37,-56 75,-56 42,0 76,33 76,75 0,76 -76,151 -151,227 -76,-76 -151,-151 -151,-227 0,-42 33,-75 75,-75 38,0 57,18 76,56z',
        matrix: [
          0.03333333333333333, 0, 0, 0.03333333333333333, -5.566666666666666,
          -5.533333333333333,
        ],
      },
    ],
    circle: ['circle'],
    star: ['star'],
  }

  const heartsOptions = (options: TCanvasConfettiAnimationOptions) => {
    return {
      ...options,
      startVelocity: 50,
      decay: 0.95,
      spread: 180,
      zIndex: 9999,
      gravity: 3,
      origin: {
        y: 1,
      },
      scalar: size || 7,
      particleCount: Math.floor(count * (Math.random() * (0.5 - 0.1) + 0.1)),
      colors: ['#AEDB01'],
      shapes: PARTICLES[particles] || ['square'],
    }
  }

  useEffect(() => {
    setTimeout(() => {
      onOnce()
    }, 1500)
  })

  return (
    <CanvaContainer>
      <Realistic onInit={onInit} decorateOptions={heartsOptions} />
    </CanvaContainer>
  )
}

export const ValentinesDayBanner = () => {
  const [isValentinesDay, setIsValentinesDay] = useState(false)

  useEffect(() => {
    const today = dayjs()
    const isFebruary14 = today.date() === 14 && today.month() === 1
    if (isFebruary14) {
      setIsValentinesDay(true)
    }
  }, [])
  return (
    <>
      {isValentinesDay && (
        <>
          <Confetti size={8} particles="heart" spreads={150} />
          <Confetti size={2} particles="heart" spreads={150} />
          <Confetti size={1} particles="heart" spreads={150} />
        </>
      )}
    </>
  )
}
