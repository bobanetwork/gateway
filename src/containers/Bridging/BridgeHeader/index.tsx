import HelpOutlineOutlined from '@mui/icons-material/HelpOutlineOutlined'
import { openModal } from 'actions/uiAction'
import { Heading } from 'components/global'
import Tooltip from 'components/tooltip/Tooltip'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { selectActiveNetworkType } from 'selectors'
import styled, { useTheme } from 'styled-components'
import { NetworkType } from 'util/network/network.util'
import { BridgeHeaderWrapper, GearIcon, IconWrapper } from './styles'
import { useNetworkInfo } from 'hooks/useNetworkInfo'

export const LabelStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 138.3%;
`

export const ValueStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 138.3%;
`

const BridgeHeader = () => {
  const dispatch = useDispatch<any>()
  const theme: any = useTheme()
  const { isSepoliaNetwork } = useNetworkInfo()
  const isTestnet =
    useSelector(selectActiveNetworkType()) === NetworkType.TESTNET

  const iconColor =
    theme.name === 'light' ? theme.colors.gray[600] : theme.colors.gray[100]

  const openSettingModal = () => {
    dispatch(openModal('settingsModal'))
  }

  const ClassicBridgeInfo = () => {
    return (
      <>
        <LabelStyle>Classic Bridge</LabelStyle> <br />
        <ValueStyle>
          Although this option is always available, it takes 7 days to receive
          your funds when withdrawing from L2 to L1.
        </ValueStyle>
        <br />
        <br />
      </>
    )
  }

  const FastBridgeInfo = () => {
    return (
      <>
        <LabelStyle>Fast Bridge</LabelStyle>
        <br />
        <ValueStyle>
          A swap-based bridge to Boba L2. This option is only available if the
          pool balance is sufficient.
        </ValueStyle>
        <br />
        <br />
      </>
    )
  }

  const LightBridgeInfo = () => {
    if (!isTestnet) {
      return <></>
    }
    return (
      <>
        {' '}
        <LabelStyle>Light Bridge</LabelStyle>
        <br />
        <ValueStyle>
          Bridge assets instantaneously and even between L2's. This option is
          only available for a few selected assets (mostly BOBA).
        </ValueStyle>
      </>
    )
  }

  return (
    <BridgeHeaderWrapper>
      <Heading variant="h2">
        Bridge
        {isSepoliaNetwork ? null : (
          <Tooltip
            data-testid="tooltip-btn"
            title={
              <>
                <ClassicBridgeInfo />
                <FastBridgeInfo />
                <LightBridgeInfo />
              </>
            }
          >
            <IconWrapper inline={true} style={{ marginLeft: '5px' }}>
              <HelpOutlineOutlined
                fontSize="small"
                sx={{ cursor: 'pointer', color: iconColor }}
              />
            </IconWrapper>
          </Tooltip>
        )}
      </Heading>
      <IconWrapper>
        <GearIcon
          sx={{ color: iconColor }}
          onClick={openSettingModal}
          data-testid="setting-btn"
        />
      </IconWrapper>
    </BridgeHeaderWrapper>
  )
}

export default BridgeHeader
