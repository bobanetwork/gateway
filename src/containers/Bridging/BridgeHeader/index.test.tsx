import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { NetworkType } from 'util/network/network.util'
import CustomThemeProvider from 'themes'
import BridgeHeader from '.'
import thunk from 'redux-thunk'
import { mockedInitialState } from 'util/tests'

const mockStore = configureStore([thunk])

const renderBridgeHeader = ({ store }: any) => {
  return render(
    <Provider store={store}>
      <CustomThemeProvider>
        <BridgeHeader />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('BridgeHeader', () => {
  let store

  beforeEach(() => {
    store = mockStore(mockedInitialState)
  })

  test('should match snapshot when connected to Mainnet', () => {
    const { asFragment } = renderBridgeHeader({ store })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should open setting modal on click of gear icon', () => {
    renderBridgeHeader({ store })
    const settingIcon = screen.getByTestId('setting-btn')
    fireEvent.click(settingIcon)
    const actions = store.getActions()
    expect(actions).toEqual([
      {
        destNetworkSelection: undefined,
        fast: undefined,
        lock: undefined,
        payload: 'settingsModal',
        proposalId: undefined,
        selectionLayer: undefined,
        token: undefined,
        tokenIndex: undefined,
        type: 'UI/MODAL/OPEN',
      },
    ])
  })

  test('should open tooltip with correct info', async () => {
    renderBridgeHeader({ store })
    const tooltipBtn = screen.getByTestId('tooltip-btn')
    fireEvent.mouseEnter(tooltipBtn)
    expect(await screen.findByText('Classic Bridge')).toBeInTheDocument()
    expect(await screen.findByText('Light Bridge')).toBeInTheDocument()
  })

  test('should match snapshot when connected to TESTNET', async () => {
    store = mockStore({
      ...mockedInitialState,
      network: {
        ...mockedInitialState.network,
        activeNetworkType: NetworkType.TESTNET,
      },
    })
    const { asFragment } = renderBridgeHeader({ store })
    expect(asFragment()).toMatchSnapshot()
  })
})
