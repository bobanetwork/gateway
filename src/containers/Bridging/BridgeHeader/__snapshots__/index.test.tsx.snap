// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BridgeHeader should match snapshot when connected to Mainnet 1`] = `
<DocumentFragment>
  <div
    class="sc-fPXMVe bdNYkh"
  >
    <h2
      class="sc-eldPxv gktDtz"
    >
      Bridge
      <div
        class="sc-gFqAkR dEURVQ"
        data-mui-internal-clone-element="true"
        data-testid="tooltip-btn"
        style="margin-left: 5px;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1t96nry-MuiSvgIcon-root"
          data-testid="HelpOutlineOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
          />
        </svg>
      </div>
    </h2>
    <div
      class="sc-gFqAkR fZrgbo"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-ikkxIA lkyGHt css-1g4na4e-MuiSvgIcon-root"
        data-testid="setting-btn"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BridgeHeader should match snapshot when connected to TESTNET 1`] = `
<DocumentFragment>
  <div
    class="sc-fPXMVe bdNYkh"
  >
    <h2
      class="sc-eldPxv gktDtz"
    >
      Bridge
      <div
        class="sc-gFqAkR dEURVQ"
        data-mui-internal-clone-element="true"
        data-testid="tooltip-btn"
        style="margin-left: 5px;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1t96nry-MuiSvgIcon-root"
          data-testid="HelpOutlineOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
          />
        </svg>
      </div>
    </h2>
    <div
      class="sc-gFqAkR fZrgbo"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-ikkxIA lkyGHt css-1g4na4e-MuiSvgIcon-root"
        data-testid="setting-btn"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
